### PR TITLE
Niak 96

### PR DESCRIPTION
**Pull request by majchrow**
Fixed Course of study POST Null Pointer bugs.
Added predecessors functionality and search history endpoint.
Added "created date" field so we can track more precisely when someone edited data.

### CODE REVIEW

File: rezerwacje-app/src/main/java/org/niewidoczniakademicy/rezerwacje/service/converter/CourseOfStudyConverter.java [hunk 1]
```diff
@@ -25,36 +23,33 @@
...
+                .orElseThrow(() ->
+                        new FacultyNotFoundException(String.format("Faculty %s not found!", dto.getFaculty())));
+
+        SystemUser opUser1 = userRepository.findByLogin(dto.getContactPerson1())
+                .orElseThrow(() ->
+                        new UserNotFoundException(String.format("User %s not found!", dto.getContactPerson1())));
+        SystemUser opUser2;
+        String opUser2Login = dto.getContactPerson2();
+        if (opUser2Login != null) {
+            opUser2 = null;
```
Comment (enthyp) [inline rezerwacje-app/src/main/java/org/niewidoczniakademicy/rezerwacje/service/converter/CourseOfStudyConverter.java:36]: I think it should be `if (opUser2Login == null) {...`

**Review by enthyp — COMMENTED** [review]
Looks good overall, just marked one tiny issue.

I also think we'll need either an additional field marking Faculty/CourseOfStudy as "current" or methods to fetch ones with the latest creationTimestamp, e.g. to fill tables on the front-end. It need not be added in this PR though.

File: rezerwacje-app/src/main/java/org/niewidoczniakademicy/rezerwacje/service/converter/CourseOfStudyConverter.java (hunk 1 above)
Comment (majchrow) [reply rezerwacje-app/src/main/java/org/niewidoczniakademicy/rezerwacje/service/converter/CourseOfStudyConverter.java:36]: Fixed

Comment (majchrow) [reply rezerwacje-app/src/main/java/org/niewidoczniakademicy/rezerwacje/service/converter/CourseOfStudyConverter.java:36]: Done

**Pull request merged**